lib/Core/Executor.cpp

```diff
@@ -1288,11 +1294,19 @@ void Executor::executeCall(ExecutionState &state,
     if (InvokeInst *ii = dyn_cast<InvokeInst>(i))
       transferToBasicBlock(ii->getNormalDest(), i->getParent(), state);
   } else {
+    // Check if maximum stack size was reached.
+    // We currently only count the number of stack frames
+    if (RuntimeMaxStackFrames && state.stack.size() > RuntimeMaxStackFrames) {
+      terminateStateOnError(state, "Stack overflow detected.", User);
```

Comment:
Here you just ran into a limit. Wouldn't it make more sense to use `terminateStateEarly` and print "Stack maximum reached"?